common/compositor/gl/glprogram.cpp

```diff
@@ -83,47 +84,54 @@ static std::string GenerateVertexShader(int layer_count) {
   return vertex_shader_stream.str();
 }
 
-static std::string GenerateFragmentShader(int layer_count) {
+static std::string GenerateFragmentShader() {
   std::ostringstream fragment_shader_stream;
   fragment_shader_stream << "#version 300 es\n"
-                         << "#define LAYER_COUNT " << layer_count << "\n"
+                         << "#define LAYER_COUNT " << MAX_LAYER_COUNT_ << "\n"
                          << "#extension GL_OES_EGL_image_external : require\n"
                          << "precision mediump float;\n";
-  for (int i = 0; i < layer_count; ++i) {
+  for (int i = 0; i < MAX_LAYER_COUNT_; ++i) {
     fragment_shader_stream << "uniform samplerExternalOES uLayerTexture" << i
                            << ";\n";
   }
   fragment_shader_stream << "uniform float uLayerAlpha[LAYER_COUNT];\n"
                          << "uniform float uLayerPremult[LAYER_COUNT];\n"
                          << "uniform vec4 uLayerColor[LAYER_COUNT];\n"
+                         << "uniform int uLayerCount;\n"
                          << "in vec2 fTexCoords[LAYER_COUNT];\n"
                          << "out vec4 oFragColor;\n"
                          << "void main() {\n"
                          << "  vec3 color = vec3(0.0, 0.0, 0.0);\n"
                          << "  float alphaCover = 1.0;\n"
                          << "  vec4 texSample;\n"
-                         << "  vec3 multRgb;\n";
-  for (int i = 0; i < layer_count; ++i) {
+                         << "  vec3 multRgb;\n"
+                         << "  int layer_index = 0;\n";
+  for (int i = 0; i < MAX_LAYER_COUNT_; ++i) {
     if (i > 0)
-      fragment_shader_stream << "  if (alphaCover > 0.5/255.0) {\n";
+      fragment_shader_stream
+          << "  if (alphaCover > 0.5/255.0 && layer_index < uLayerCount) {\n";
+    else
+      fragment_shader_stream << "  if(layer_index < uLayerCount) {\n";
```

Comment:
this line is reducdant,
if i==0 layer_index will be less then uLayerCount for sure